repository/src/main/java/org/apache/atlas/accesscontrol/persona/AtlasPersonaService.java

```diff
@@ -0,0 +1,599 @@
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("updatePersona");
+        LOG.info("Updating Persona");
+
+        AtlasEntity persona = context.getPersona();
+
+        AtlasEntity existingPersonaEntity = existingPersonaWithExtInfo.getEntity();
+
+        try {
+            if (!AtlasEntity.Status.ACTIVE.equals(existingPersonaEntity.getStatus())) {
+                throw new AtlasBaseException(OPERATION_NOT_SUPPORTED, "Persona not Active");
+            }
+
+            if (getPersonaRoleId(persona) != getPersonaRoleId(existingPersonaEntity)) {
+                throw new AtlasBaseException(ATTRIBUTE_UPDATE_NOT_SUPPORTED, PERSONA_ENTITY_TYPE, ATTR_RANGER_ROLE_ID);
+            }
+
+            if (getIsEnabled(existingPersonaEntity) != getIsEnabled(persona)) {
+                if (getIsEnabled(context.getPersona())) {
+                    enablePersona(existingPersonaWithExtInfo);
+                } else {
+                    disablePersona(existingPersonaWithExtInfo);
+                }
+            }
+
+            if (!getName(persona).equals(getName(existingPersonaEntity))) {
+                validateUniquenessByName(graph, getName(persona), PERSONA_ENTITY_TYPE);
+            }
+
+            atlasRangerService.updateRangerRole(context);
+
+            aliasStore.updateAlias(context);
+        } finally {
+            RequestContext.get().endMetricRecord(metricRecorder);
+        }
+    }
+
+    public void deletePersona(AtlasEntityWithExtInfo personaExtInfo) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("deletePersona");
+        AtlasEntity persona = personaExtInfo.getEntity();
+
+        try {
+            if(!persona.getTypeName().equals(PERSONA_ENTITY_TYPE)) {
+                throw new AtlasBaseException(BAD_REQUEST, "Please provide entity of type " + PERSONA_ENTITY_TYPE);
+            }
+
+            if(!persona.getStatus().equals(AtlasEntity.Status.ACTIVE)) {
+                LOG.info("Persona with guid {} is already deleted/purged", persona.getGuid());
+                return;
+            }
+
+            cleanRoleFromAllRangerPolicies(personaExtInfo);
+
+            atlasRangerService.deleteRangerRole(getPersonaRoleId(persona));
+
+            aliasStore.deleteAlias(getESAliasName(persona));
+
+        } finally {
+            RequestContext.get().endMetricRecord(metricRecorder);
+        }
+    }
+
+    private void cleanRoleFromAllRangerPolicies(AtlasEntityWithExtInfo personaExtInfo) throws AtlasBaseException {
+        AtlasEntity persona = personaExtInfo.getEntity();
+
+        List<RangerPolicy> rangerPolicies = fetchRangerPoliciesByLabel(atlasRangerService,
+                null,
+                null,
+                getPersonaLabel(persona.getGuid()));
+
+        List<String> allPolicyGuids = getPolicies(personaExtInfo).stream()
```

Comment:
these are policy labels right and not policyGuids?

Reply:
need to rename fields